packages/lit-dev-content/src/github/github-signin-receiver-page.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import type {GitHubSigninReceiverMessage} from './github-types.js';
+
+// This is the inlined script for the /playground/signin/ page.
+//
+// This page is assumed to be running in a popup window created by the main
+// Playground page. After the user clicks "Accept" or "Cancel" on the GitHub
+// authorization page, GitHub will redirect to this page with a "code" or
+// "error" URL parameter. This page will then message the original parent
+// Playground window with that code or error, and the popup window will close.
+
+const opener = window.opener as Window | null;
+if (opener) {
+  const url = new URL(document.location.href);
+  const params = url.searchParams;
+  const code = params.get('code');
+  const error = params.get('error');
+  let message: GitHubSigninReceiverMessage;
+  if (error) {
+    message = {error};
+  } else if (!code) {
+    message = {error: 'No code'};
+  } else {
+    message = {code};
+  }
+  opener.postMessage(message);
```

Comment:
Is there also a way to notify the `opener` if the page is closed? Since this is a mock it may not be relevant.

Reply:
Nice! Also thanks for clarifying how to popup will evolve!